### ISSUE TITLE
Move from lerna to npm 7 workspaces

### ISSUE THREAD

**Issue by ojeytonwilliams** [freeCodeCamp/freeCodeCamp#41261]
**Is your feature request related to a problem? Please describe.**
With npm 7 we can use [workspaces](https://docs.npmjs.com/cli/v7/using-npm/workspaces) to manage sub-packages.

**Describe the solution you'd like**
Since we have to use a package manager, but may not have to use lerna it would be nice to reduce the number of dependencies. Assuming that workspaces can meet our needs.

**Describe alternatives you've considered**
Continue using lerna, since we know it works.

**Comment by nhcarrigan** [comment]
Linking #41210 - `lockfileversion 2` rolled out in `npm` v7.

**Comment by raisedadead** [comment]
I was wondering if its better we explored the workspaces instead of https://github.com/freeCodeCamp/freeCodeCamp/pull/41210? Would it make sense to wait and kill two birds at the same time if this pans out well?

**Comment by ojeytonwilliams** [comment]
Well, the justification for that PR is supporting npm 6. If we use workspaces, we can just make npm 7 a requirement. So, sure, I can look into workspaces.

**Comment by raisedadead** [comment]
Yeah, that would be ideal actually.

**Comment by nhcarrigan** [comment]
Has anyone tested `npm` 7 on `node` 12?

I've been using `node` 14, myself, to see if there are any conflicts with our codebase.

**Comment by ojeytonwilliams** [comment]
I've tested `npm 7` with `node 12` and it seems fine.

As for workspaces, I've spent an hour tinkering with them and I don't think they do what we want, yet.  Since the api is going to be deployed independently, it needs its own `package-lock.json` that's a) self-contained and b) what we use in dev, CI etc.

That doesn't seem possible with workspaces.  Instead of creating individual locks for each package, it creates one root lock file with all that information inside it.  

If you `npm i` in the root, it'll generate the root lock file, but subsequent installs like

```
npm ci package-you-want
```

installs each package and

```
cd package-you-want
npm ci
```

just fails because the lock doesn't exist in that dir.

Unless I'm missing something, I think we should stick with `lerna` for now and keep an eye on how workspaces develop.
